src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -430,8 +437,8 @@ partial class BoundLiteral : ILiteralExpression
 
     partial class BoundObjectCreationExpression : IObjectCreationExpression
     {
-        private static readonly ConditionalWeakTable<BoundObjectCreationExpression, object> s_memberInitializersMappings =
-            new ConditionalWeakTable<BoundObjectCreationExpression, object>();
+        private static readonly ConditionalWeakTable<BoundObjectCreationExpression, StrongBox<ImmutableArray<IMemberInitializer>>> s_memberInitializersMappings =
+            new ConditionalWeakTable<BoundObjectCreationExpression, StrongBox<ImmutableArray<IMemberInitializer>>>();
 
```

Comment:
I agree that in terms of memory allocation, there may not be much difference. Given that, we should avoid using a less familiar type.